Library/Homebrew/keg.rb

```diff
@@ -297,12 +297,18 @@ def lock
   def completion_installed?(shell)
     dir = case shell
     when :bash then path.join("etc", "bash_completion.d")
-    when :zsh  then path.join("share", "zsh", "site-functions")
+    when :zsh  then path.join("share", "zsh", "site-functions") if path.join("share", "zsh", "site-functions").children.any? { |f| f.basename.to_s.start_with?("_") }.empty?
```

Comment:
I feel like there should be a cleaner way to do this.
Any Ruby devs have a suggestion?